### PR TITLE
GEODE-3998: Links Cache::Name to DistributeSystem::Name

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**

### CODE REVIEW

File: cppcache/src/CacheFactory.cpp [hunk 1]
```diff
@@ -109,39 +132,6 @@ Cache CacheFactory::create() const {
...
-      cache.close();
-    }
-    throw apache::geode::client::UnknownException(
-        "Exception thrown in CacheFactory::create");
-  }
-
-  return cache;
-}
-
 std::shared_ptr<CacheFactory> CacheFactory::set(const std::string& name,
```
Comment (echobravopapa) [inline cppcache/src/CacheFactory.cpp]: If Cache name is linked to DS name, doesn't this need to flow through to the DS...?

Comment (pivotal-jbarrett) [reply cppcache/src/CacheFactory.cpp]: Yes, but I am not sure what you are referring to here?

Comment (echobravopapa) [reply cppcache/src/CacheFactory.cpp]: CacheFactory::set() takes a name parameter, which is inserted into the dsProperties... is that guaranteed to update the name the DS uses?

Comment (pivotal-jbarrett) [reply cppcache/src/CacheFactory.cpp]: This method takes a name/value pair to update ds properties. It is not explicitly setting the "ds name". 

The fact that this is the worst way to handle configuration is outside the scope of this issue.

Comment (echobravopapa) [reply cppcache/src/CacheFactory.cpp]: fair enough...

**Pull request merged**